From 7bf2e4d5ca1c94a9b0f730498b3d01768a72dcbd Mon Sep 17 00:00:00 2001
Message-Id: <7bf2e4d5ca1c94a9b0f730498b3d01768a72dcbd.1651196605.git.geliang.tang@suse.com>
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Fri, 7 Jan 2022 10:34:07 +0300
Subject: [PATCH] ACPI: pfr_telemetry: Fix info leak in pfrt_log_ioctl()
Git-commit: 7bf2e4d5ca1c94a9b0f730498b3d01768a72dcbd
Patch-mainline: v5.17-rc1
References: bsn#210
Modified-by-SEL: No


The "data_info" struct is copied to the user.  It has a 4 byte struct
hole after the last struct member so we need to memset that to avoid
copying uninitialized stack data to the user.

Fixes: b0013e037a8b ("ACPI: Introduce Platform Firmware Runtime Telemetry driver")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/acpi/pfr_telemetry.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/acpi/pfr_telemetry.c b/drivers/acpi/pfr_telemetry.c
index da50dd80192c..9abf350bd7a5 100644
--- a/drivers/acpi/pfr_telemetry.c
+++ b/drivers/acpi/pfr_telemetry.c
@@ -83,6 +83,7 @@ static int get_pfrt_log_data_info(struct pfrt_log_data_info *data_info,
 	union acpi_object *out_obj, in_obj, in_buf;
 	int ret = -EBUSY;
 
+	memset(data_info, 0, sizeof(*data_info));
 	memset(&in_obj, 0, sizeof(in_obj));
 	memset(&in_buf, 0, sizeof(in_buf));
 	in_obj.type = ACPI_TYPE_PACKAGE;
-- 
2.34.1

